sdk/nodejs/output.ts

```diff
@@ -124,24 +124,26 @@ class OutputImpl<T> implements OutputInstance<T> {
         return utils.isInstance(obj, "__pulumiOutput");
     }
 
+    /** @internal */
+    static async getPromisedValue<T>(promise: Promise<T>, withUnknowns?: boolean): Promise<T> {
+        // If the caller did not explicitly ask to see unknown values and val contains unknowns, return undefined. This
+        // preserves compatibility with earlier versions of the Pulumi SDK.
+        const val = await promise;
+        if (!withUnknowns && containsUnknowns(val)) {
+            return <T><any>undefined;
+        }
+        return val;
+    }
+
     /** @internal */
     public constructor(
             resources: Set<Resource> | Resource[] | Resource,
             promise: Promise<T>,
             isKnown: Promise<boolean>,
             isSecret: Promise<boolean>) {
 
-        const value = Promise.all([isKnown, promise]).then(([known, val]) => (withUnknowns?: boolean) => {
-            // If the caller did not explicitly ask to see unknown values, simply return val.
-            if (!withUnknowns) {
-                return containsUnknowns(val) ? <T><any>undefined : val;
-            }
-
-            return val;
-        });
-
-        // Otherwise, we are only known if we are not explicitly unknown and the resolved value of the output
-        // contains no distinguished unknown values and if the value's promise is not rejected.
+        // We are only known if we are not explicitly unknown and the resolved value of the output
+        // contains no distinguished unknown values.
         isKnown = Promise.all([isKnown, promise]).then(([known, val]) => known && !containsUnknowns(val));
```

Comment:
note: my extractoin comment also referred to this guy.  it would make it very clear what flows in/out.

Reply:
I don't think that it's worth pulling this one out, especially since its value is immediately assigned to a field.